services/src/contexts/migrations/migration_0000_initial.rs

```diff
@@ -0,0 +1,105 @@
+use async_trait::async_trait;
+use tokio_postgres::Transaction;
+
+use crate::{
+    error::Result,
+    layers::{
+        add_from_directory::UNSORTED_COLLECTION_ID, storage::INTERNAL_LAYER_DB_ROOT_COLLECTION_ID,
+    },
+};
+
+use super::database_migration::{DatabaseVersion, Migration};
+
+pub struct Migration0000Initial;
+
+#[async_trait]
+impl Migration for Migration0000Initial {
+    fn prev_version(&self) -> Option<DatabaseVersion> {
+        None
+    }
+
+    fn version(&self) -> DatabaseVersion {
+        "0000_initial".into()
+    }
+
+    async fn migrate(&self, tx: &Transaction<'_>) -> Result<()> {
+        let config = crate::util::config::get_config_element::<crate::util::config::Postgres>()?;
+
+        let schema_name = &config.schema;
+
+        if schema_name != "pg_temp" {
+            tx.batch_execute(&format!("CREATE SCHEMA IF NOT EXISTS {schema_name};",))
+                .await?;
+        }
+
+        tx.batch_execute(include_str!("migration_0000_initial.sql"))
+            .await?;
+
+        let stmt = tx
+            .prepare(
+                "
+            INSERT INTO geoengine (clear_database_on_start, database_version) VALUES ($1, '0000_initial');",
+            )
+            .await?;
```

Comment:
the `migrate_database` method checks this. I think it's better to do it at one place, because an individual migration could forget to check it. Now it is automatically checked.

The version is also only inserted in the initial migration. The migration logic in `migrate_database` updates the current version automatically after each migration.

Reply:
Could you point to the place where it automatically updates it?

Reply:
https://github.com/geo-engine/geoengine/pull/881/files#diff-5f7b6dfc9a5e3b93a8b875257f9f86c8c9e138aba9864ac9e166f9cc39e26e4bR108